p2p/src/sync/peer.rs

```diff
@@ -149,8 +151,8 @@ where
     }
 
     async fn main_loop(&mut self) -> Result<()> {
-        let mut stalling_interval = tokio::time::interval(*self.p2p_config.sync_stalling_timeout);
-        stalling_interval.set_missed_tick_behavior(MissedTickBehavior::Delay);
+        let mut periodic_check = tokio::time::interval(*self.p2p_config.sync_stalling_timeout);
```

Comment:
I find stalling_interval (or `peer_stalling_interval`) a more appropriate name. Is there a reason we're renaming this?

Generic names like `period_check` are very bad as they cannot be understood in generic contexts.